js/src/common/helpers/listItems.tsx

```diff
@@ -3,13 +3,13 @@ import Separator from '../components/Separator';
 import classList from '../utils/classList';
 import type * as Component from '../Component';
 
-function isSeparator(item): boolean {
+function isSeparator(item: Mithril.Vnode): boolean {
   return item.tag === Separator;
 }
 
-function withoutUnnecessarySeparators(items: Array<Mithril.Vnode>): Array<Mithril.Vnode> {
-  const newItems = [];
-  let prevItem;
+function withoutUnnecessarySeparators(items: Mithril.Vnode[]): Mithril.Vnode[] {
```

Comment:
Should this be `Vnode[]` or `Children`?

Reply:
Either should work to pass into the function.

```suggestion
function withoutUnnecessarySeparators(items: Mithril.Children): Mithril.Children {
```